code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -267,6 +272,8 @@
 	if(!revealed) //Revenants cannot die if they aren't revealed
 		return 0
 	..(1)
+	ghost_darkness_images -= ghostimage
+	updateallghostimages()
```

Comment:
You should put this below line 286 so that it's still visible to ghosts with darkness off for the 1~ second it's dying.